# ⚗️ Melanie's screen GEMs

## 📕 Description

This repo contains inputs, outputs, and step-by-step details regarding how models were reconstructed for strains used in Melanie's screen. For more details check out the [SymbNET metabolic modeling tutorial](https://github.com/franciscozorrilla/SymbNET).

## 🧱 Materials

* `genomes` DNA fasta files for 40 NCBI strains used in experimental screen
* `models` Genome scale metabolic models gapfilled on various media
* `media` Growth media formulations used for gapfilling and simulation
* `proteomes` Open reading frame (ORF)-annotated protein fasta files
* `ensembles` Ensemble models for network uncertainty quantification
* `notebooks` Scripts used to visualize resulting models
* `plots` Figures generated by notebooks  

## 🐪 Software used

* git
* Prodigal
* CarveMe
   * diamond
   * CPLEX

## 🩺 Methods

### 0. Clone repo

```
$ git clone https://github.com/franciscozorrilla/melanie_screen_GEMs.git
```

### 1. Translate genomes to ORF-annotated protein fasta files using prodigal

Move into cloned repository directory and create `proteomes` folder

```
$ cd melanie_screen_GEMs
$ mkdir -p proteomes
```

Run prodigal on each input genome file found in the `genomes/` folder

```
$ while read file; do prodigal -i genomes/$file -a proteomes/${file%.*}.faa;done< <(ls genomes/)
```

### 2. Create genome scale metabolic models using CarveMe 

Output models with fbc2 format gapfilling on M3 media

```
$ mkdir -p models/M3_gapfilled
$ while read model;do carve -v --mediadb media/media_db.tsv -g M3 --fbc2 -o models/M3_gapfilled/${model%.*}.xml proteomes/$model; done< <(ls proteomes)
```

Output models with fbc2 format without gapfilling
```
$ mkdir -p models/no_gapfill
$ while read model;do carve -v --fbc2 -o models/no_gapfill/${model%.*}.xml proteomes/$model; done< <(ls proteomes/)
```

### 3. Create ensemble models

```

```

## 🏌️‍♂️ Results
